src/main/kotlin/reactor/kotlin/extra/math/MathFluxExtensions.kt

```diff
@@ -115,11 +424,66 @@ fun <T> Flux<T>.sumDouble(mapper: (T) -> Number): Mono<Double>
  *
  * @param mapper a lambda converting values to [Number]
  * @author Simon Baslé
- * @since 3.1.1
+ * @since 1.0.0
+ * @deprecated Please use averageDouble() as a direct replacement of consider more general purpose averageAll()
  */
 fun <T> Flux<T>.average(mapper: (T) -> Number): Mono<Double>
         = MathFlux.averageDouble(this, Function(mapper))
 
+/**
+ * Extension to map arbitrary values in a [Flux] to [Number]s and return the average of
+ * these Numbers as a [Mono] of [Float].
+ *
+ * Note that numbers will be mapped to [Float] using Java standard conversions
+ * This may lead to arbitrary precision gain or loss if the source flux contains different types
+ * Please choose an appropriate method based on the expected types in the source flux.
+ *
+ * @param mapper a lambda converting values to [Number]
+ * @author Mark Pruden
+ * @since 1.1.5
+ */
+fun <T> Flux<T>.averageFloat(mapper: (T) -> Number): Mono<Float>
```

Comment:
as for sumTypes(mapper), averageTypes(mapper) should not be included in this PR

Reply:
Have removed these methods.